### PR TITLE
pytorch and fastrcnn

### PR DESCRIPTION
**Pull request by xzdandy**
Add a new integration test cases for data/ua_detrac.mp4 and src/udfs/fastrcnn_object_detector.py.

Tested on Ada-01 with GPU. 

Add gpu_batch_size in eva.yml

### CODE REVIEW

File: src/udfs/pytorch_abstract_udf.py [hunk 1]
```diff
@@ -43,21 +44,25 @@ def get_device(self):
...
+    def transform(self, images: np.ndarray):
+        # reverse the channels from opencv
+        return self.transforms(Image.fromarray(images[:, :, ::-1]))\
+            .unsqueeze(0)
 
-    def forward(self, frames: np.ndarray):
-        return self.classify([self.transform(frame[0]) for frame in frames])
+    def forward(self, frames: List[np.ndarray]):
+        tens_batch = torch.cat([self.transform(x) for x in frames])\
+            .to(self.get_device())
```
Comment (gaurav274) [inline src/udfs/pytorch_abstract_udf.py:54]: Is this done to speed up the data transfer?

**Review by gaurav274 — COMMENTED** [review]
Travis won't run a GPU test so we might have to skip it on the master build

File: src/udfs/pytorch_abstract_udf.py (hunk 1 above)
Comment (xzdandy) [reply src/udfs/pytorch_abstract_udf.py:54]: This is batched (multiple frames) model for pytorch. But From the latency numbers I get.  I do not see performance improvement.

**Comment by xzdandy** [comment]
Hopefully, gpu_batch_size 1 can work on travis with cpu version of pytorch.

File: src/udfs/fastrcnn_object_detector.py [hunk 2]
```diff
@@ -105,12 +105,25 @@ def _get_predictions(self, frames: Tensor) -> List[Outcome]:
...
+                ignore_index=True)
+        return outcome
+
+    def classify(self, frames: Tensor) -> pd.DataFrame:
+        gpu_batch_size = ConfigurationManager()\
+            .get_value('executor', 'gpu_batch_size')
 
-    def classify(self, frames: Tensor) -> List[Outcome]:
-        return self._get_predictions(frames)
+        if gpu_batch_size:
```
Comment (gaurav274) [inline src/udfs/fastrcnn_object_detector.py:121]: Can we move this to the abstract class so that it happens automatically for every UDF?

Comment (xzdandy) [reply src/udfs/fastrcnn_object_detector.py]: Done in fe148be

**Review by gaurav274 — APPROVED** [review]

**Pull request merged**